utils/shared.js

```diff
@@ -757,6 +757,18 @@ const isDateTimeFormat = (value) => {
     return typeof value == "string" && (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value));
 }
 
+const redactEmailLoginInfo = (loginDetails) => {
+    let amp = loginDetails.indexOf('@');    
+    for(let i = 0; i < amp; i++) {
+        if(i != 0 && i != 1 && i != amp - 1) {
+            loginDetails = loginDetails.substring(0, i) + "*" + loginDetails.substring(i + 1);
+        } 
+    }
+    return loginDetails
```

Comment:
Nice refactor for the email and phone redacting logic. First time seeing a the redacting of characters for email address.Interesting way of redacting the characters before the "@" in the email address.

Reply:
I'm doing the same part for speed. And avoided "for" loop:
```js
  const [prefix, domain] = fetchedData[421823980].split("@");
  const changedPrefix = prefix.length > 3
      ? prefix.slice(0, 2) + "*".repeat(prefix.length - 3) + prefix.slice(-1)
      : prefix.slice(0, -1) + "*";
  loginDetails = changedPrefix + "@" + domain;
```
We do have participants whose emails only have 2 letters beside @ and domain.